zk-sdk/src/sigma_proofs/percentage_with_cap.rs

```diff
@@ -0,0 +1,703 @@
+//! The percentage-with-cap sigma proof.
+//!
+//! The proof is defined with respect to three Pedersen commitments that encodes values referred to
+//! as a `percentage`, `delta`, and `claimed` amounts. The proof certifies that either
+//! - the `percentage` amount is equal to a constant (referred to as the `max_value`)
+//! - the `delta` and `claimed` amounts are equal
+//!
+//! This type of proof is useful as a building block to prove that, given two Pedersen commitments,
+//! one encodes a percentage value of the number encoded by the other commitment with a specified
+//! max cap value.
+//!
+//! A more detailed description of the context and how the proof is computed is provided in the
+//! [`ZK Token proof program`] documentation.
+//!
+//! The protocol guarantees computational soundness (by the hardness of discrete log) and perfect
+//! zero-knowledge in the random oracle model.
+//!
+//! [`ZK Token proof program`]: https://docs.solanalabs.com/runtime/zk-token-proof
```

Comment:
I guess you'll go back through in future work to rename zk-token-proof to zk-proof, correct?

Reply:
Yep, those have to be updated. Will do on a follow-up!